tests/event.rs

```diff
@@ -0,0 +1,280 @@
+use std::collections::{BTreeSet, HashSet};
```

Comment:
In general, I would prefer this to live at the bottom of the module it's testing, but that's just personal preference.